src/Common/src/CoreLib/System/ReadOnlySpan.cs

```diff
@@ -83,13 +83,13 @@ public override int GetHashCode()
         /// <summary>
         /// Defines an implicit conversion of an array to a <see cref="ReadOnlySpan{T}"/>
         /// </summary>
-        public static implicit operator ReadOnlySpan<T>(T[] array) => array != null ? new ReadOnlySpan<T>(array) : default;
+        public static implicit operator ReadOnlySpan<T>(T[] array) => new ReadOnlySpan<T>(array);
 
         /// <summary>
         /// Defines an implicit conversion of a <see cref="ArraySegment{T}"/> to a <see cref="ReadOnlySpan{T}"/>
         /// </summary>
         public static implicit operator ReadOnlySpan<T>(ArraySegment<T> arraySegment)
-            => arraySegment.Array != null ? new ReadOnlySpan<T>(arraySegment.Array, arraySegment.Offset, arraySegment.Count) : default;
+            => new ReadOnlySpan<T>(arraySegment.Array, arraySegment.Offset, arraySegment.Count);
```

Comment:
Is it possible to have an ArraySegment for a `null` array with nontrivial bounds? Or do we care?

Reply:
You can't construct an array segment with a null array (it throws: https://msdn.microsoft.com/en-us/library/911cay97(v=vs.110).aspx)

One way the arraysegment.Array could be null is as follows, but in that case offset/count are 0 anyway:
```C#
ArraySegment<byte> segment = default;
Console.WriteLine(segment.Offset); // 0
Console.WriteLine(segment.Count); // 0
```

Reply:
Makes sense. Then passing it to the ctor seems fine. 
Still need to decide whether to do with range checks.